### PR TITLE
added the start of a style guide.

### PR DESCRIPTION
**Pull request by ChrisBarker-NOAA**
Here's a start -- please review and we can then merge and start pointing to it from pysgrid and pyugrid.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@ChrisBarker-NOAA I will merge this so I can start editing it.

Thanks for starting the document!

**Comment by ChrisBarker-NOAA** [comment]
thanks -- could you add in your preferences for quotations?

-CHB

File: style_guide.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                 grid_topology_var=None,
+                 variables=None,
+                 grid_variables=None,
+                 dimensions=None,
+                 node_dimensions=None,
+                 node_coordinates=None,
+                 edge1_coordinates=None,
+                 edge2_coordinates=None,
+                 angles=None,
+                 )
```
Comment (rhattersley) [inline style_guide.rst:69]: If your function has this many parameters maybe formatting isn't the problem that needs solving? :wink:

Comment (ocefpaf) [inline style_guide.rst:69]: @rhattersley has a point. (And that discussion should be in `pysgrid` as this was copied from there.)

Comment (PythonCHB) [inline style_guide.rst:69]: Ironically, I truncated it -- there are quite a few more ;-)

Though I don't have a problem with an initializer with all the (optional)
parameters it needs to fully initialize the object.

And there are other cases where a lot of optional parameters make sense.

In any case, too much to fit on one line is not rare -- hence the style
suggestion.

On Thursday, March 10, 2016, Filipe notifications@github.com wrote:

> In style_guide.rst
> https://github.com/ioos/APIRUS/pull/18#discussion_r55664711:
> 
> > -                 center_lon=None,
> > -                 center_lat=None,
> > -                 edges=None,
> > -                 node_padding=None,
> > -                 edge1_padding=None,
> > -                 edge2_padding=None,
> > -                 grid_topology_var=None,
> > -                 variables=None,
> > -                 grid_variables=None,
> > -                 dimensions=None,
> > -                 node_dimensions=None,
> > -                 node_coordinates=None,
> > -                 edge1_coordinates=None,
> > -                 edge2_coordinates=None,
> > -                 angles=None,
> > -                 )
> 
> @rhattersley https://github.com/rhattersley has a point. (And that
> discussion should be in pysgrid as this was copied from there.)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ioos/APIRUS/pull/18/files#r55664711.

## 

Christopher Barker, PhD

Python Language Consulting
- Teaching
- Scientific Software Development
- Desktop GUI and Web Development
- wxPython, numpy, scipy, Cython

**Pull request merged**